pyo3-macros-backend/src/attrs.rs

```diff
@@ -0,0 +1,27 @@
+use proc_macro2::TokenStream;
+use quote::ToTokens;
+use syn::spanned::Spanned;
+use syn::{Meta, MetaNameValue, Result};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct FromPyWithAttribute(syn::ExprPath);
+
+impl FromPyWithAttribute {
+    pub fn from_meta(meta: Meta) -> Result<Self> {
+        if let Meta::NameValue(MetaNameValue {
+            lit: syn::Lit::Str(string_literal),
+            ..
+        }) = meta
+        {
+            let expr_path = string_literal.parse::<syn::ExprPath>()?;
+            return Ok(FromPyWithAttribute(expr_path));
+        }
+        bail_spanned!(meta.span() => "expected a name-value: `pyo3(from_py_with = \"func\")` ")
+    }
+}
+
+impl ToTokens for FromPyWithAttribute {
```

Comment:
Curious what this impl is needed for?

Reply:
Since I'd wrapped ExprPath in a struct, I couldn't use it in `quote!` without getting the nested `ExprPath` by index:
```rust
match attrs.from_py_with {
    Some(expr) => quote!(#expr.0),
    ...
}
```

I didn't like how `#expr.0` looked, so I proxied the`ToTokens` of the nested `ExprPath`. It became possible to use `FromPyWithAttribute` directly:
```rust
match attrs.from_py_with {
    Some(expr) => quote!(#expr),
    ...
}
```

Only now I see that it's not actually necessary, we can match for the nested ExprPath instead:

```rust
match attrs.from_py_with {
    Some(FromPyWithAttribute(expr)) => quote!(#expr),
    ...
}
```